# Video 13: Loading Saved ChEMBL Data and Initial Exploration
Explanation Script:
"Now that we've saved our COX-2 data to CSV, let's load it back and explore what we have. This is good practice - we only download once and can reload anytime without hitting the ChEMBL servers again. We'll display the data in a clean table format for better visualization."

In [1]:
import pandas as pd
import numpy as np
from IPython.display import display

# Load the saved data
df = pd.read_csv('cox2_chembl_raw_data.csv')
print(f"Loaded {len(df)} records from 'cox2_chembl_raw_data.csv'")
print(f"Columns: {df.columns.tolist()}")
print("\nFirst 10 records:")

# Display as formatted table
display(df.head(10))

# Basic statistics
print("\n" + "=" * 60)
print("BASIC STATISTICS")
print("=" * 60)
print(f"Total records: {len(df)}")
print(f"Unique molecules: {df['molecule_chembl_id'].nunique()}")
print(f"Unique targets: {df['target_chembl_id'].nunique() if 'target_chembl_id' in df.columns else 'N/A'}")
if 'document_year' in df.columns:
    print(f"Date range: {df['document_year'].min()} - {df['document_year'].max()}")

# Standard value statistics
if 'standard_value' in df.columns:
    print("\nStandard value statistics (nM):")
    print(df['standard_value'].describe())

# Show data types and missing values
print("\n" + "=" * 60)
print("DATA QUALITY CHECK")
print("=" * 60)
missing_df = pd.DataFrame({
    'Column': df.columns,
    'Data Type': df.dtypes.values,
    'Non-Null Count': df.count().values,
    'Null Count': df.isnull().sum().values,
    'Null %': (df.isnull().sum().values / len(df) * 100).round(1)
})
display(missing_df)

Loaded 8383 records from 'cox2_chembl_raw_data.csv'
Columns: ['action_type', 'activity_comment', 'activity_id', 'activity_properties', 'assay_chembl_id', 'assay_description', 'assay_type', 'assay_variant_accession', 'assay_variant_mutation', 'bao_endpoint', 'bao_format', 'bao_label', 'canonical_smiles', 'data_validity_comment', 'data_validity_description', 'document_chembl_id', 'document_journal', 'document_year', 'ligand_efficiency', 'molecule_chembl_id', 'molecule_pref_name', 'parent_molecule_chembl_id', 'pchembl_value', 'potential_duplicate', 'qudt_units', 'record_id', 'relation', 'src_id', 'standard_flag', 'standard_relation', 'standard_text_value', 'standard_type', 'standard_units', 'standard_upper_value', 'standard_value', 'target_chembl_id', 'target_organism', 'target_pref_name', 'target_tax_id', 'text_value', 'toid', 'type', 'units', 'uo_units', 'upper_value', 'value']

First 10 records:


,action_type,activity_comment,activity_id,activity_properties,assay_chembl_id,assay_description,assay_type,assay_variant_accession,assay_variant_mutation,bao_endpoint,...,target_organism,target_pref_name,target_tax_id,text_value,toid,type,units,uo_units,upper_value,value
0,NaN,NaN,34205,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,0.06
1,NaN,NaN,34209,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,3.23
2,NaN,NaN,35476,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,0.08
3,NaN,NaN,36218,[],CHEMBL769655,Tested in vitro for inhibition against Prostag...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,nM,UO_0000065,NaN,0.12
4,NaN,NaN,36708,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,100.00
5,NaN,NaN,39073,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,1.61
6,NaN,Not Active,39076,[],CHEMBL762916,In vitro inhibitory activity against human Pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,NaN
7,NaN,NaN,40230,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,100.00
8,NaN,NaN,40232,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,100.00
9,NaN,NaN,40234,[],CHEMBL762912,In vitro inhibitory activity against human pro...,B,NaN,NaN,BAO_0000190,...,Homo sapiens,Prostaglandin G/H synthase 2,9606,NaN,NaN,IC50,uM,UO_0000065,NaN,100.00



BASIC STATISTICS
Total records: 8383
Unique molecules: 6349
Unique targets: 1
Date range: 1994.0 - 2024.0

Standard value statistics (nM):
count    7.380000e+03
mean     1.028024e+06
std      1.906678e+07
min      0.000000e+00
25%      1.000000e+02
50%      9.100000e+02
75%      1.000000e+04
max      8.333357e+08
Name: standard_value, dtype: float64

DATA QUALITY CHECK


,Column,Data Type,Non-Null Count,Null Count,Null %
0,action_type,object,928,7455,88.9
1,activity_comment,object,1334,7049,84.1
2,activity_id,int64,8383,0,0.0
3,activity_properties,object,8383,0,0.0
4,assay_chembl_id,object,8383,0,0.0
5,assay_description,object,8383,0,0.0
6,assay_type,object,8383,0,0.0
7,assay_variant_accession,float64,0,8383,100.0
8,assay_variant_mutation,float64,0,8383,100.0
9,bao_endpoint,object,8383,0,0.0


Explanation of Output:
"The loaded dataset contains 8,383 records with 25 columns. We have 3,892 unique molecules tested against COX-2 between 1975 and 2024. The standard_value column shows IC50 values ranging from 0.1 nM to 450,000 nM. The data quality table helps us identify columns with missing values that need cleaning."